app/meteor-accounts-saml/server/saml_server.js

```diff
@@ -268,52 +268,112 @@ const middleware = function(req, res, next) {
 			case 'logout':
 				// This is where we receive SAML LogoutResponse
 				_saml = new SAML(service);
-				_saml.validateLogoutResponse(req.query.SAMLResponse, function(err, result) {
-					if (!err) {
-						const logOutUser = function(inResponseTo) {
-							if (Accounts.saml.settings.debug) {
-								console.log(`Logging Out user via inResponseTo ${ inResponseTo }`);
-							}
+				if (req.query.SAMLRequest) {
+					_saml.validateLogoutRequest(req.query.SAMLRequest, function(err, result) {
+						if (err) {
+							console.error(err);
+							throw new Meteor.Error('Unable to Validate Logout Request');
+						}
+
+						const logOutUser = function(samlInfo) {
 							const loggedOutUser = Meteor.users.find({
-								'services.saml.inResponseTo': inResponseTo,
+								$or: [
+									{ 'services.saml.nameID': samlInfo.nameID },
+									{ 'services.saml.idpSession': samlInfo.idpSession },
+								],
 							}).fetch();
+
 							if (loggedOutUser.length === 1) {
-								if (Accounts.saml.settings.debug) {
-									console.log(`Found user ${ loggedOutUser[0]._id }`);
-								}
 								Meteor.users.update({
```

Comment:
this update is now duplicated, I guess it could be have its own function to generalize it.

Reply:
moved it to its own little function